compiler/src/main/java/bullet/impl/ComponentProcessingStep.java

```diff
@@ -152,15 +166,33 @@ private String getName(TypeMirror type) {
             .build())
         .addModifiers(PUBLIC, FINAL)
         .addSuperinterface(ClassName.get("bullet", "ObjectGraph"))
-
         .addField(elementName, "component", PRIVATE, FINAL)
-
         .addMethod(MethodSpec.constructorBuilder()
             .addModifiers(PUBLIC)
             .addParameter(elementName, "component", FINAL)
             .addCode("this.component = component;\n")
             .build());
 
+    if (membersInjectionMethodsMap.size() > 0) {
+      classBuilder.addField(ClassName.get("bullet.impl", "ClassIndexHashTable"), "classIndexHashTable", PRIVATE, STATIC, FINAL);
+    }
+
+    // Generate the ClassIndexHashTable if there are classes to inject.
+    if (membersInjectionMethodsMap.size() > 0) {
```

Comment:
That second `if` is not needed: you could just merge the two `if`s together as they have the same condition.